omniscidb/Tests/QueryBuilderTest.cpp

```diff
@@ -493,6 +509,28 @@ TEST_F(QueryBuilderTest, Arithmetics) {
   compare_res_data(res, std::vector<int64_t>({0, NULL_BIGINT}));
 }
 
+TEST_F(QueryBuilderTest, DateToInt) {
+  QueryBuilder builder(ctx(), schema_mgr_, configPtr());
+
+  auto tinfo_a = builder.scan("test_date");
+
+  auto dag = tinfo_a.proj(tinfo_a.ref("col_date").cast("int32")).finalize();
+  auto res = runQuery(std::move(dag));
+  LOG(ERROR) << "res: " << res.toString();
+  LOG(ERROR) << "res: " << toArrow(res)->ToString();
+  compare_res_data(res, std::vector<int32_t>({946684800, 1426636800}));
```

Comment:
Are these numbers what you would expect for used dates in `kDay` time unit?

Reply:
I converted `2000-01-01, 2015-03-18` to int and used this values. (also checked what happens with online converter https://timestamp.online) In general it can be expanded to Hours and Milli.

Reply:
This converter gives you a timestamp in seconds. I would expect different int values when casting dates in different units.